### DISCUSSION TITLE
Nice work! Any chance to have a power sensor?

### DISCUSSION THREAD

**Discussion by pihiandreas** [echavet/MitsubishiCN105ESPHome#141] [Ideas]
First thanks to all contributors for an exceptionally good component! 

I can confirm that this works with the MSZ-LN25VG2.

I've been working mostly with Panasonic ACs and a similar esphome-component (DomiStyle) and I'm wondering does the Mitsubishi´s also collect power usage (Watt) and could this sensor be added in this component? 

Also, I would assume that the Mitsubishi also collects the external temperature (from the outside unit) for de-icing scheduling etc, is this data really not available over the CN105?

**Comment by echavet** [comment]
Hi @pihiandreas,
It is probably possible but the protocol isn't public. So we have to discover how.
At this time, we don't know how to get the temperature from the external unit, and we don't know anything about the watt consumption.
We only know, for supported units, the compressor frequency and it's activity.

**Reply by pihiandreas** [reply]
OK, thanks, I'll try to dig into the CN105 code ...

**Reply by phidauex** [reply]
To my knowledge none of the units report power or energy consumption, which is annoying (the Mitsubishi control board is certainly monitoring this). A few people here have been able to identify a few other elements, like the auto status, the fan speed, and compressor frequency (only some units support this), but no power data. One check would be to see what Kumo Cloud would make available for your unit - if it isn't supported there, then it probably isn't available on the CN105 interface.

**Comment by phidauex** [comment]
A power and energy sensor were added in https://github.com/echavet/MitsubishiCN105ESPHome/pull/166 - they don't work on every unit, but you might get lucky!

**Reply by pihiandreas** [reply]
Nice 🤩🤩🤩 Will test it out asap..
